src/process/ProcessManagerImpl.cpp

```diff
@@ -511,12 +511,25 @@ ProcessExitEvent::Impl::run()
     // Iterate through all possibly open file descriptors except stdin, stdout,
     // and stderr and set FD_CLOEXEC so the subprocess doesn't inherit them
     const int maxFds = sysconf(_SC_OPEN_MAX);
-    for (int fd = 3; fd < maxFds; ++fd)
+    // as the space of open file descriptors is arbitrary large
+    // we use as a heuristic the number of consecutive unused descriptors
+    // as an indication that we're past the range where descriptors are
+    // allocated
+    // a better way would be to enumerate the opened descriptors, but there
+    // doesn't seem to be a portable way to do this
+    const int maxGAP = 512;
```

Comment:
any particular reason it's 512?

Reply:
Technically, is it possible to hit a weird edge case where FD_CLOEXEC is not set for some fds?

Reply:
It's a heuristic:
* Linux will reuse descriptors starting from small numbers (filling gaps), so it concentrates descriptors in the bottom range
* we only need to set `FD_CLOEXEC` once (which limits the chance that we miss high descriptors)
* 512 is a number much larger than the churn that typically happens on file descriptors (a "gap" will appear when closing 512 connections all at once, and not reconnect)

If we miss a descriptor it's not that big of a deal, it will cause some connections to stay alive longer (client side will kill them).